File: workspaces/aware_kernel/modules/api/libs/api/dart/lib/src/context.dart
import 'package:uuid/uuid.dart';

class AwareApiContext {
  const AwareApiContext({
    required this.environmentId,
    this.processId,
    this.threadId,
    this.branchId,
    this.projectionHash,
    this.actorId,
  });

  final UuidValue environmentId;
  final UuidValue? processId;
  final UuidValue? threadId;
  final UuidValue? branchId;
  final String? projectionHash;
  final UuidValue? actorId;

  AwareApiContext copyWith({
    UuidValue? environmentId,
    UuidValue? processId,
    UuidValue? threadId,
    UuidValue? branchId,
    String? projectionHash,
    UuidValue? actorId,
  }) {
    return AwareApiContext(
      environmentId: environmentId ?? this.environmentId,
      processId: processId ?? this.processId,
      threadId: threadId ?? this.threadId,
      branchId: branchId ?? this.branchId,
      projectionHash: projectionHash ?? this.projectionHash,
      actorId: actorId ?? this.actorId,
    );
  }
}

File: workspaces/aware_network/modules/api/libs/api/python/aware_api/context.py
"""Small transport context DTO shared by endpoint-invocation callers."""

from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, Field


class AwareApiContext(BaseModel):
    """Runtime lane context carried by callers that already know their session."""

    environment_id: UUID
    process_id: UUID
    thread_id: UUID
    branch_id: UUID | None = Field(default=None)
    projection_hash: str | None = Field(default=None)


__all__ = ["AwareApiContext"]
